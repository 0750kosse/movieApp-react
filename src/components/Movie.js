import React, { Component } from 'react'

const url = "http://image.tmdb.org/t/p/w185/"

class Movie extends Component {
  state = {
    singleMovie: {}
  }

  componentDidMount() {
    let id = this.props.match.params.movie_id;
    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=96fc446ecb31737673fc835496bd6ead&language=en-US&include_adult=false`)
      .then(res => res.json())
      .then(data => this.setState({ singleMovie: data }))
  }

  render() {
    const { id, poster_path, title } = this.state.singleMovie

    return (<div className="movies-container">
      <div className="movie-card__single">
        <img src={`${url}` + poster_path} alt="poster" className="movie-poster__single"></img>
        <p >This is the id: {id}</p>
        <p >This is the title: {title}</p>
      </div>
    </div>)
  }
}

export default Movie;


