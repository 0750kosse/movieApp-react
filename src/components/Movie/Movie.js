import React, { Component } from 'react'

const url = "http://image.tmdb.org/t/p/w185/"

class Movie extends Component {
  state = {
    singleMovie: {}
  }

  componentDidMount() {
    let id = this.props.match.params.movie_id;
    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=96fc446ecb31737673fc835496bd6ead&language=en-US&include_adult=false`)
      .then(res => res.json())
      .then(data => this.setState({ singleMovie: data }))
  }
  render() {
    const { id, poster_path, overview, title, vote_average, release_date, runtime } = this.state.singleMovie
    return (
      <div className="movies-container">
        <div className="movie-card__single" key={id}>
          <h2>{title}</h2>
          <img src={`${url}` + poster_path} alt="poster" className="movie-poster__single"></img>
          <div className="movie-details">
            <p>Overview: {overview}</p>
            <p>Runtime: {runtime} min</p>
            <p>Rating: {vote_average}</p>
            <p>Year: {release_date}</p>
          </div>
        </div>
      </div>
    )
  }
}


export default Movie;


