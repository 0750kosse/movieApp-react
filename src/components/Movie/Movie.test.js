import React from 'react'
import Movie from './Movie'
import { shallow, mount } from 'enzyme';

describe('Movie component', () => {

  it('renders correctly', () => {
    let componentDidMountSpy = spyOn(Movie.prototype, 'componentDidMount')
    const wrapper = shallow(<Movie />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call componentDidMount once', () => {
    let componentDidMountSpy = spyOn(Movie.prototype, 'componentDidMount')
    let wrapper = shallow(<Movie />)
    expect(componentDidMountSpy).toHaveBeenCalledTimes(1)
  })
})