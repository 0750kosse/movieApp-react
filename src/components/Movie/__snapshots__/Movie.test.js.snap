// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component renders correctly 1`] = `ShallowWrapper {}`;
